docs/data-sources.md

```diff
@@ -0,0 +1,40 @@
+
+## Gecko Profiler (aka SPS, nsIProfiler)
+
+The profiler collects two types of information, samples and markers. The profiler is an internal component inside of Gecko. It stores all of profile information in a circular buffer, that when it gets full, the buffer starts to rewrite the old data with new data. This is a nice feature because it means the profiler can be left on indefinitely, allowing for the easy capture of a profile once some kind of interesting behavior happens.
+
+### Samples in the profiler
+
+Samples are taken of the current executing function at a fixed interval, e.g. every 1 millisecond. These samples are collected and provide a general statistical view of where time was spent executing code. The more a function is executed, the more samples show up in the buffer. These samples include a variety of data including their call stacks, and information about the function name. In JavaScript a string is provided with the function name, the file location, and the line number. C++ functions only contain their memory address. Cleopatra then must take the additional step of "symbolicating" these memory addresses, by looking up the original function names in a function table that is specific to that individual build of Firefox.
+
+### Markers in the profiler
+
+Samples don't record every event that happens within the system, so some information gets lost. Markers on the other hand, get fired every time certain events happen within the system. These events can be arbitrarily added across Gecko by engineers wherever they may wish. This can be useful for exposing expensive operations that may be missed by the profiler, and the markers provide a precise view of how various systems are running.
+
+## Timeline - DocShell Markers
+
+The Gecko Profiler records marker data, but it doesn't include all of the markers available in the system. There is a competing implementation of markers used exclusively by the Firefox Devtools that are recorded per DocShell. These are not currently integrated into Cleopatra.
+
+* [Timeline C++ Implementation](https://dxr.mozilla.org/mozilla-central/source/docshell/base/timeline)
+* [Timeline Devtools JS Server](https://dxr.mozilla.org/mozilla-central/source/devtools/server/performance/timeline.js)
+
+## Task Tracer
+
+TaskTracer provides a way to trace the correlation between different tasks across threads and processes. Unlike sampling based profilers, TaskTracer can tell you where a task is dispatched from, what its original source was, how long it waited in the event queue, and how long it took to execute.
+
+Source Events are usually some kinds of I/O events we're interested in, such as touch events, timer events, network events, etc. When a source event is created, TaskTracer records the entire chain of Tasks and nsRunnables as they are dispatched to different threads and processes. It records latency, execution time, etc. for each Task and nsRunnable that chains back to the original source event.
```

Comment:
The Gecko C++ code here has changed a little in the meantime - the "Task" class is no longer used, and "nsRunnable" has been renamed to mozilla::Runnable. So I'd replace "Task or nsRunnable" with "Runnable".